File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int c=0;
        for(int i=0;i<=s.size();i++){
            int l=i,r=i;
            while (l>=0 and r<s.size() and s[l]==s[r]){
                c++;
                l--;
                r++;
                
            }

            l=i;
            r=i+1;
            while (l>=0 and r<s.size() and s[l]==s[r]){
                c++;
                l--;
                r++;
                
            }
        }
        return c;


    }
};

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        ans=0
        stack=[]
        
        for i,h in enumerate(heights):
            extensionindex=i
            while stack and h<stack[-1][1]:
                startindex=stack[-1][0]
                area=(i-startindex)*stack[-1][1]
                ans=max(ans,area)
                extensionindex=startindex
                stack.pop()
            
            stack.append((extensionindex,h))
        while stack:
            i,h=stack[-1]
            area=(len(heights)-i)*h
            ans=max(ans,area)
            stack.pop()
        return ans